### ISSUE TITLE
OpenMS documentation

### ISSUE THREAD

**Issue by hroest** [OpenMS/OpenMS#3557]
I would like to open a general discussion about the OpenMS documentation (after #3553). I noticed several issues

- documentation is rather spread out (which are partially overlapping):
  - wiki
  - doxygen
  - tutorial https://github.com/OpenMS/Tutorials
- it is not that easy for developers to figure out where to start 
- some parts are really outdated, such as https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/nightly/html/known_dev_bugs.html
- it currently does not look really great (compare https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/nightly/html/index.html with http://seqan.readthedocs.io/en/master/index.html#)
- it is currently self-hosted while other projects like seqan use readthedocs.io (that is not a problem per-se, but another task for our developers to think about)

**Comment by timosachsenberg** [comment]
Hi Hannes, I think you raised some valid points.

- documentation is rather spread out (which are partially overlapping): wiki, doxygen, tutorial
Agreed, I think we should reduce developer/user documentation overlap between wiki and doxygen. Should also be rather quick to fix. Regarding the tutorials, we separated them because they also come with workflows/data etc. we didn't want to pollute the main repository with. They also are usually tailored to tutorial sessions (see the different branches) which complicates things a bit. 

- it is not that easy for developers to figure out where to start
The README.MD points to the documentation which also contains a quite extensive (still covering only a smart part of OpenMS) developer tutorial.  Probably some extra confusions arise from duplicated / redundant parts in the wiki? What would you like to improve here?

- some parts are really outdated
I even wasn't aware of this page. Let's throw them out. Such things are nicely covered by GitHub issue system.

- Doesn't look too great, I agreed. IIRC they actually got extra funding to work on the tutorial to get it into that shape.

**Comment by timosachsenberg** [comment]
@jpfeuffer 
http://www.openms.de/current_doxygen/
currently links to
http://ftp.mi.fu-berlin.de/pub/OpenMS/release-documentation/html/index.html
which is really outdated and not the current documentation. We probably need to fix this on the OpenMS.de webpage.

**Comment by hroest** [comment]
I think a big part is that we now have a nice user tutorial and a user can figure out where to get started, but our developer tutorial is not really streamlined nor as up to date. I think the thing to improve would be a single tutorial that explains how to install OpenMS, where to get help, how the internals work and how to write your own tool. We have all the bits and pieces, its mostly putting them together that is missing and making it clear from the README with a big red button (blinking?) "click here"? I have noticed this since I started working on a tutorial for pyOpeMS and I think it could go along with the OpenMS tutorial hand in hand (most code is the same between Python and C++ -- who knew?). 

Maybe we should just clarify which stuff should live where. E.g. I think doxygen is and will be the API documentation, the question is whether faster-evolving stuff should live somewhere else. We have partly decided this already as some stuff has moved to the wiki where we can evolve faster and quicker but we probably need a rule where to find what. 

Again, this is also in the context of where the pyOpenMS tutorial and documentation should live.

**Comment by timosachsenberg** [comment]
https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/nightly/html/OpenMS_tutorial.html
By the way - in the nightlies we have the developer tutorial we recently added. Don’t know if you already saw it. I think we could build on this.
